### PR TITLE
Restructure adding and removing policies from users and groups

### LINKED ISSUE #4272: Modify admin policy command
Change the `add` sub-command to `create`

Output of `mc admin policy --help` will be as follows.
```
create     create a new IAM policy
list          list all IAM policies
info        show info on an IAM policy
remove  remove an IAM policy
```
Also, add examples to create a policy and assign it to a user/group in the help of `mc admin policy create`

Also, `set`, `unset` and `upate` sub-commands will be removed and moved under `mc admin user policy` and `mc admin group policy` commands

### LINKED ISSUE #4273: Add sub-commands under admin user policy
Add sub-commands under `mc admin user policy`

`mc admin user policy [attach | detach | list]`
```
attach     attach an IAM policy to a user
detach   detach an IAM policy from a user
list  list all IAM policies attached to a user
```
Also, add examples to create a user and attach a policy to the user. This needs to be done as part of help under `mc admin user policy` and `mc admin user add`

### LINKED ISSUE #4274: Add sub-commands under admin group policy
Add sub-commands under  `mc admin group policy`

`mc admin group policy [attach | detach | list]`
```
attach     attach an IAM policy to a group
detach   detach an IAM policy from a group
list  list all IAM policy attached to a group
```
Also, add examples to create a group and attach a policy to the group. This needs to be done as part of help under mc admin group policy and mc admin group add

### PR DESCRIPTION
**Pull request by taran-p**
## Description
Changes `mc admin policy add` to `mc admin policy create`
Replaces `mc admin policy (set|unset|update)` commands with `mc admin policy (attach|detach)`
Adds `mc admin policy entities`

## Motivation and Context
Implements changes requested in issues below, with some modifications requested offline.

resolves #4272 
resolves #4273 
resolves #4274 

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [x] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fmc%2fpull%2fNNNNN)

### CODE REVIEW

**Review by harshavardhana — APPROVED** [review]
This is a breaking change FWIW

**Comment by harshavardhana** [comment]
PTAL @donatello

File: cmd/admin-policy-unset.go [hunk 1]
```diff
@@ -18,120 +18,22 @@
...
-			op:          ctx.Command.Name,
-			Policy:      policiesToUnset,
-			UserOrGroup: userOrGroup,
-			IsGroup:     isGroup,
-		})
-	} else {
-		fatalIf(probe.NewError(e), "Unable to unset the policy")
-	}
+func mainAdminPolicyUnsetErr(ctx *cli.Context) error {
+	console.Infoln("Please use 'mc admin policy detach'")
```
Comment (donatello) [inline cmd/admin-policy-unset.go:37]: This function should return an error, and ensure that mc exits with non-zero exit status.

File: cmd/admin-policy-set.go [hunk 2]
```diff
@@ -18,102 +18,22 @@
...
-			op:          ctx.Command.Name,
-			Policy:      policyName,
-			UserOrGroup: userOrGroup,
-			IsGroup:     isGroup,
-		})
-	} else {
-		fatalIf(probe.NewError(e), "Unable to set the policy")
-	}
+func mainAdminPolicySetErr(ctx *cli.Context) error {
+	console.Infoln("Please use 'mc admin policy attach'")
```
Comment (donatello) [inline cmd/admin-policy-set.go:37]: This function should return an error, and ensure that mc exits with non-zero exit status.

File: cmd/admin-policy-update.go [hunk 3]
```diff
@@ -18,125 +18,22 @@
...
-			op:          ctx.Command.Name,
-			Policy:      policiesToAdd,
-			UserOrGroup: userOrGroup,
-			IsGroup:     isGroup,
-		})
-	} else {
-		fatalIf(probe.NewError(e), "Unable to unset the policy")
-	}
+func mainAdminPolicyUpdateErr(ctx *cli.Context) error {
+	console.Infoln("Please use 'mc admin policy attach'")
```
[the remark this answers is not in the document]
Comment (taran-p) [reply cmd/admin-policy-update.go:37]: This is the formatting for other commands that are no longer in use  , but I realise now that those commands don't end in err. Would it make more sense to change the command name to `mainAdminPolicyUpdate` but keep the formatting consistent or change it in a different way?

**Review by donatello — APPROVED** [review]

**Comment by donatello** [comment]
PTAL @harshavardhana @kannappanr this is ready to merge

This is a breaking change and we should note it in the release notes in the next release.

**Pull request merged**